http/src/request/scheduled_event/update_guild_scheduled_event.rs

```diff
@@ -0,0 +1,212 @@
+    entity_type: Option<EntityType>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    name: Option<&'a str>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    privacy_level: Option<PrivacyLevel>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    scheduled_end_time: Option<NullableField<&'a Timestamp>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    scheduled_start_time: Option<&'a Timestamp>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    status: Option<Status>,
+}
+
+/// Update a scheduled event in a guild.
+///
+/// This endpoint supports changing the type of event. When changing the entity
+/// type to either [`EntityType::StageInstance`] or [`EntityType::Voice`], an
+/// [`Id<ChannelMarker>`] must be provided if it does not already exist.
+///
+/// When changing the entity type to [`EntityType::External`], the `channel_id`
+/// field is cleared and the [`channel_id`] method has no effect.  Additionally,
+/// you must set a location with [`location`].
+///
+/// [`channel_id`]: UpdateGuildScheduledEvent::channel_id
+/// [`location`]: UpdateGuildScheduledEvent::location
+/// [`channel_id`]: UpdateGuildScheduledEvent::channel_id
+/// [`location`]: UpdateGuildScheduledEvent::location
+#[must_use = "requests must be configured and executed"]
+pub struct UpdateGuildScheduledEvent<'a> {
+    guild_id: Id<GuildMarker>,
+    http: &'a Client,
+    fields: UpdateGuildScheduledEventFields<'a>,
+    reason: Option<&'a str>,
+    scheduled_event_id: Id<ScheduledEventMarker>,
+}
+
+impl<'a> UpdateGuildScheduledEvent<'a> {
+    pub(crate) const fn new(
+        http: &'a Client,
+        guild_id: Id<GuildMarker>,
+        scheduled_event_id: Id<ScheduledEventMarker>,
+    ) -> Self {
+        Self {
+            guild_id,
+            http,
+            fields: UpdateGuildScheduledEventFields {
+                channel_id: None,
+                description: None,
+                entity_metadata: None,
+                entity_type: None,
+                name: None,
+                privacy_level: None,
+                scheduled_end_time: None,
+                scheduled_start_time: None,
+                status: None,
+            },
+            reason: None,
+            scheduled_event_id,
+        }
+    }
+
+    /// Set the channel ID.
+    ///
+    /// If `entity_type` is already [`EntityType::External`], this has no
+    /// effect.
+    pub fn channel_id(mut self, channel_id: Id<ChannelMarker>) -> Self {
+        if let Some(entity_type) = self.fields.entity_type {
+            if entity_type == EntityType::External {
+                return self;
+            }
+        }
+
+        self.fields.channel_id = Some(NullableField(Some(channel_id)));
+
+        self
+    }
+
+    /// Set the description of the event.
+    ///
+    /// Must be between 1 and 1000 characters in length.
+    pub const fn description(mut self, description: Option<&'a str>) -> Self {
```

Comment:
There must always be a description
```suggestion
    pub const fn description(mut self, description: &'a str) -> Self {
```

Reply:
<img width="489" alt="image" src="https://user-images.githubusercontent.com/1760581/147806924-4d495b9c-4734-49e8-9c14-4153051857c2.png">

Reply:
https://discord.com/developers/docs/resources/guild-scheduled-event#guild-scheduled-event-object-guild-scheduled-event-structure